### PR TITLE
Fixed issue with reboot detection and reutrn of get method in xSPInstallPrereqs

### PR DESCRIPTION
**Pull request by BrianFarnhill**
@caadam @camiloborges @anlynes @ykuijs - Code review, feedback on prereq installer

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xsharepoint/246)

### CODE REVIEW

File: Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1 [hunk 1]
```diff
@@ -81,15 +81,27 @@ function Set-TargetResource
...
-    else
-    {
-        throw "SharePoint install failed, exit code was $($setup.ExitCode)"
+    switch ($setup.ExitCode) {
+        0 {  
+            Write-Verbose -Message "SharePoint binary installation complete"
+            $global:DSCMachineStatus = 1
+        }
+        30066 {
+            if (    ((Get-Item 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending' -ErrorAction SilentlyContinue) -ne $null) `
```
Comment (camiloborges) [inline Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1]: Is there a module to provide this validation? If not, should we wrap this as a function inside .Util?

**Comment by BrianFarnhill** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1, line 90 \[r1\]](https://reviewable.io:443/reviews/powershell/xsharepoint/246#-KE4HKynuFLME_bIExiH-r1-90:-KEYFsbLOMhlHj-Re9X0:-857755463) ([raw file](https://github.com/powershell/xsharepoint/blob/407e036685c021a5024ca75b61992c54ec4ff051/Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1#L90)):*
xPendingReboot has some of this logic in it, but for the sake of one line it was easy enough to integrate in to this one spot we have here. Given these are the only places we are ever going to need to trigger a reboot from keeping it inside the modules seemed safe enough. Happy to leave it here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsharepoint/246)*

**Comment by camiloborges** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1, line 90 \[r1\]](https://reviewable.io:443/reviews/powershell/xsharepoint/246#-KE4HKynuFLME_bIExiH-r1-90:-KEYSfBoQJYvplIA1mzh:-1372840659) ([raw file](https://github.com/powershell/xsharepoint/blob/407e036685c021a5024ca75b61992c54ec4ff051/Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1#L90)):*
Trying to close and publish

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsharepoint/246)*

**Comment by camiloborges** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsharepoint/246)*

**Pull request merged**